### PR TITLE
`Development`: Use id selectors in cypress e2e tests for text exercises

### PR DESCRIPTION
**Pull request by jhainzlschmid**
### Motivation and Context

To improve the stability of the e2e test suite we want to only use id selectors to match UI elements.

### Description

In this PR I refactor the existing cypress tests for text exercises (and the used page objects) to not use css selectors or selectors which depend on a translation which comes from the Artemis Client.

### Steps for Testing

[The tests are executed by Bamboo and still pass](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-694)

#### Code Review
- [x] Review 1
- [ ] Review 2

### CODE REVIEW

File: src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts [hunk 1]
```diff
@@ -23,30 +23,22 @@ export class TextEditorPage {
...
+        cy.get('#participation-header').contains(exerciseTitle).should('be.visible');
     }
 
     shouldShowProblemStatement() {
-        cy.get('[jhitranslate="artemisApp.exercise.problemStatement"]').should('be.visible');
+        cy.get('#problem-statement').should('be.visible');
     }
 
     getHeaderElement() {
         return cy.get('jhi-header-participation-page');
```
Comment (rs-maker) [inline src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts:34]: This isnt an id.

File: src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts [hunk 2]
```diff
@@ -55,8 +55,8 @@ export class TextExerciseCreationPage {
         cy.get('[jhitranslate="artemisApp.modelingExercise.createExampleSubmission"]').click();
```
Comment (rs-maker) [inline src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts:55]: This isnt an id either.

**Review by rs-maker — CHANGES_REQUESTED** [review]
Mostly ok i left two comments where possible and mention the remaining selectors in the list below.

TextExerciseAssessmentPage.ts
- line 9: not an id but also not used at all. is this an artifact? 
TextEditorPage.ts:
- line 7: two selectors that are not ids
- line 15: two selectors that are not ids
- line 21: one selector that is not an id
- line 34: not an id

File: src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts (hunk 1 above)
Comment (jhainzlschmid) [reply src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts:34]: Correct this method isn't used in a test anymore, so I missed it

File: src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts (hunk 2 above)
Comment (jhainzlschmid) [reply src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts:55]: Same here the method isn't used anymore

File: src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts (hunk 1 above)
Comment (jhainzlschmid) [reply src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts]: I removed it

File: src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts (hunk 2 above)
Comment (jhainzlschmid) [reply src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts]: I removed it

**Comment by jhainzlschmid** [comment]
> Mostly ok i left two comments where possible and mention the remaining selectors in the list below.
> 
> TextExerciseAssessmentPage.ts
> 
> * line 9: not an id but also not used at all. is this an artifact?

Yes I removed it

> TextEditorPage.ts:
> 
> * line 7: two selectors that are not ids

Why two selectors? I replaced the one selector with an id.

> * line 15: two selectors that are not ids

This code is not used by the text exercise specs. Its only used in an exam related spec, so I will refactor it there

> * line 21: one selector that is not an id

Good catch, I missed it. Should be fine now

> * line 34: not an id

Do you mean the `contains` command? I replaced it with a direct assertion, but in some cases this might be more cumbersome than using `contains`

**Comment by jhainzlschmid** [comment]
[Tests still pass after the changes](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-700)z

**Review by rs-maker — APPROVED** [review]
👍 As far as i can tell this is great now.

**Pull request merged**